### PR TITLE
feat: Rollup

### PR DESCRIPTION
**Pull request by gkatsev**

### CODE REVIEW

**Comment by imbcmdth** [comment]
This smells ok to me...

File: scripts/modules.rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export default {
+  moduleName: 'm3u8-parser',
+  entry: 'src/index.js',
+  legacy: true,
+  plugins: [
+    json(),
+    babel({
+      babelrc: false,
+      exclude: 'node_modules/**',
+      presets: [
```
Comment (gesinger) [inline scripts/modules.rollup.config.js]: This may need es3

File: scripts/test.rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }),
+    json(),
+    commonjs({
+      sourceMap: false
+    }),
+    babel({
+      babelrc: false,
+      exclude: 'node_modules/**',
+      presets: [
+        ['es2015', {
```
Comment (gesinger) [inline scripts/test.rollup.config.js]: Only tests, but maybe es3 for consistency

File: scripts/umd.rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }),
+    json(),
+    commonjs({
+      sourceMap: false
+    }),
+    babel({
+      babelrc: false,
+      exclude: 'node_modules/**',
+      presets: [
+        ['es2015', {
```
Comment (gesinger) [inline scripts/umd.rollup.config.js]: This may need es3

File: test/m3u8.test.js [hunk 4]
```diff
@@ -68,14 +68,14 @@ QUnit.test('stops sending events after deregistering', function() {
   this.lineStream.on('data', temporary);
   this.lineStream.on('data', permanent);
   this.lineStream.push('line one\n');
-  QUnit.strictEqual(temporaryLines.length,
+  assert.strictEqual(temporaryLines.length,
                     permanentLines.length,
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 5]
```diff
@@ -94,13 +94,13 @@ QUnit.test('parses comment lines', function() {
   });
   this.lineStream.push(manifest);
 
-  QUnit.ok(element, 'an event was triggered');
-  QUnit.strictEqual(element.type, 'comment', 'the type is comment');
-  QUnit.strictEqual(element.text,
+  assert.ok(element, 'an event was triggered');
+  assert.strictEqual(element.type, 'comment', 'the type is comment');
+  assert.strictEqual(element.text,
                     manifest.slice(1, manifest.length - 1),
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 6]
```diff
@@ -124,15 +124,15 @@ QUnit.test('parses unknown tag types', function() {
   });
   this.lineStream.push(manifest);
 
-  QUnit.ok(element, 'an event was triggered');
-  QUnit.strictEqual(element.type, 'tag', 'the type is tag');
-  QUnit.strictEqual(element.data,
+  assert.ok(element, 'an event was triggered');
+  assert.strictEqual(element.type, 'tag', 'the type is tag');
+  assert.strictEqual(element.data,
                     manifest.slice(4, manifest.length - 1),
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 7]
```diff
@@ -193,15 +193,15 @@ QUnit.test('parses #EXTINF tags with a duration and title', function() {
...
-  QUnit.strictEqual(element.type, 'tag', 'the line type is tag');
-  QUnit.strictEqual(element.tagType, 'inf', 'the tag type is inf');
-  QUnit.strictEqual(element.duration, 13, 'the duration is parsed');
-  QUnit.strictEqual(element.title,
+  assert.ok(element, 'an event was triggered');
+  assert.strictEqual(element.type, 'tag', 'the line type is tag');
+  assert.strictEqual(element.tagType, 'inf', 'the tag type is inf');
+  assert.strictEqual(element.duration, 13, 'the duration is parsed');
+  assert.strictEqual(element.title,
                     manifest.substring(manifest.indexOf(',') + 1, manifest.length - 1),
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 8]
```diff
@@ -210,17 +210,17 @@ QUnit.test('parses #EXTINF tags with carriage returns', function() {
...
-  QUnit.strictEqual(element.type, 'tag', 'the line type is tag');
-  QUnit.strictEqual(element.tagType, 'inf', 'the tag type is inf');
-  QUnit.strictEqual(element.duration, 13, 'the duration is parsed');
-  QUnit.strictEqual(element.title,
+  assert.ok(element, 'an event was triggered');
+  assert.strictEqual(element.type, 'tag', 'the line type is tag');
+  assert.strictEqual(element.tagType, 'inf', 'the tag type is inf');
+  assert.strictEqual(element.duration, 13, 'the duration is parsed');
+  assert.strictEqual(element.title,
                     manifest.substring(manifest.indexOf(',') + 1, manifest.length - 2),
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 9]
```diff
@@ -549,39 +549,39 @@ QUnit.test('parses #EXT-X-STREAM-INF with common attributes', function() {
...
 
-  QUnit.ok(element, 'an event was triggered');
-  QUnit.strictEqual(element.type, 'tag', 'the line type is tag');
-  QUnit.strictEqual(element.tagType, 'stream-inf', 'the tag type is stream-inf');
-  QUnit.strictEqual(element.attributes.CODECS,
+  assert.ok(element, 'an event was triggered');
+  assert.strictEqual(element.type, 'tag', 'the line type is tag');
+  assert.strictEqual(element.tagType, 'stream-inf', 'the tag type is stream-inf');
+  assert.strictEqual(element.attributes.CODECS,
                     'avc1.4d400d, mp4a.40.2',
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 10]
```diff
@@ -590,17 +590,17 @@ QUnit.test('parses #EXT-X-STREAM-INF with arbitrary attributes', function() {
...
-  QUnit.strictEqual(element.type, 'tag', 'the line type is tag');
-  QUnit.strictEqual(element.tagType, 'stream-inf', 'the tag type is stream-inf');
-  QUnit.strictEqual(element.attributes.NUMERIC, '24', 'numeric attributes are parsed');
-  QUnit.strictEqual(element.attributes.ALPHA,
+  assert.ok(element, 'an event was triggered');
+  assert.strictEqual(element.type, 'tag', 'the line type is tag');
+  assert.strictEqual(element.tagType, 'stream-inf', 'the tag type is stream-inf');
+  assert.strictEqual(element.attributes.NUMERIC, '24', 'numeric attributes are parsed');
+  assert.strictEqual(element.attributes.ALPHA,
                     'Value',
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 11]
```diff
@@ -683,27 +683,27 @@ QUnit.test('parses lightly-broken #EXT-X-KEY tags', function() {
   });
   this.lineStream.push(manifest);
 
-  QUnit.strictEqual(element.attributes.URI,
+  assert.strictEqual(element.attributes.URI,
                     'https://example.com/single-quote',
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 12]
```diff
@@ -683,27 +683,27 @@ QUnit.test('parses lightly-broken #EXT-X-KEY tags', function() {
...
                     'parsed a single-quoted uri');
 
   element = null;
   manifest = '#EXT-X-KEYURI="https://example.com/key",METHOD=AES-128\n';
   this.lineStream.push(manifest);
-  QUnit.strictEqual(element.tagType, 'key', 'parsed the tag type');
-  QUnit.strictEqual(element.attributes.URI,
+  assert.strictEqual(element.tagType, 'key', 'parsed the tag type');
+  assert.strictEqual(element.attributes.URI,
                     'https://example.com/key',
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 13]
```diff
@@ -683,27 +683,27 @@ QUnit.test('parses lightly-broken #EXT-X-KEY tags', function() {
...
                     'inferred a colon after the tag type');
 
   element = null;
   manifest = '#EXT-X-KEY:  URI =  "https://example.com/key",METHOD=AES-128\n';
   this.lineStream.push(manifest);
-  QUnit.strictEqual(element.attributes.URI,
+  assert.strictEqual(element.attributes.URI,
                     'https://example.com/key',
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 14]
```diff
@@ -849,12 +849,12 @@ QUnit.test('parses characteristics attribute', function() {
 
   parser.push(manifest);
 
-  QUnit.equal(parser.manifest.mediaGroups.SUBTITLES.subs.test.characteristics,
+  assert.equal(parser.manifest.mediaGroups.SUBTITLES.subs.test.characteristics,
               'char',
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 15]
```diff
@@ -866,21 +866,21 @@ QUnit.test('parses FORCED attribute', function() {
 
   parser.push(manifest);
 
-  QUnit.ok(parser.manifest.mediaGroups.SUBTITLES.subs.test.forced,
+  assert.ok(parser.manifest.mediaGroups.SUBTITLES.subs.test.forced,
            'parsed FORCED attribute');
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: test/m3u8.test.js [hunk 16]
```diff
@@ -866,21 +866,21 @@ QUnit.test('parses FORCED attribute', function() {
...
   let key;
 
   for (key in testDataManifests) {
     if (testDataExpected[key]) {
       const parser = new Parser();
 
       parser.push(testDataManifests[key]);
-      QUnit.deepEqual(parser.manifest,
+      assert.deepEqual(parser.manifest,
                       testDataExpected[key],
```
Comment (gesinger) [inline test/m3u8.test.js]: minor: indentation

File: scripts/modules.rollup.config.js (hunk 1 above)
Comment (gkatsev) [inline scripts/modules.rollup.config.js]: added

**Pull request merged**